code/game/objects/structures/mirror.dm

```diff
@@ -36,44 +73,149 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/structure/mirror, 28)
+		if(CHANGE_RACE)
+			change_race(user)
+		if(CHANGE_SEX) // sex: yes
+			change_sex(user)
+		if(CHANGE_NAME)
+			change_name(user)
+		if(CHANGE_EYES)
+			change_eyes(user)
+
+	return what_is_your_wish_masta(user)
+
+/obj/structure/mirror/proc/change_beard(mob/living/carbon/human/beardresser)
+	if(beardresser.physique != FEMALE && !magical_mirror)
+		var/new_style = tgui_input_list(beardresser, "Select a facial hairstyle", "Grooming", GLOB.facial_hairstyles_list)
 		if(isnull(new_style))
 			return TRUE
-		if(!user.can_perform_action(src, FORBID_TELEKINESIS_REACH))
-			return TRUE //no tele-grooming
-		if(HAS_TRAIT(hairdresser, TRAIT_SHAVED))
-			to_chat(hairdresser, span_notice("If only growing back facial hair were that easy for you..."))
+		if(HAS_TRAIT(beardresser, TRAIT_SHAVED))
+			to_chat(beardresser, span_notice("If only growing back facial hair were that easy for you... The reminder makes you feel terrible."))
+			beardresser.add_mood_event("bald_hair_day", /datum/mood_event/bald)
 			return TRUE
-		hairdresser.set_facial_hairstyle(new_style, update = TRUE)
+		beardresser.set_facial_hairstyle(new_style, update = TRUE)
 	else
-		hairdresser.set_facial_hairstyle("Shaved", update = TRUE)
+		if(beardresser.facial_hairstyle == "Shaved")
+			to_chat(beardresser, span_notice("You look towards your facial hair for a make-up and realize you don't have any."))
+			return
+		beardresser.set_facial_hairstyle("Shaved", update = TRUE)
 
-	//handle normal hair
-	var/new_style = tgui_input_list(user, "Select a hairstyle", "Grooming", GLOB.hairstyles_list)
+/obj/structure/mirror/proc/change_hair(mob/living/carbon/human/hairdresser)
+	var/new_style = tgui_input_list(hairdresser, "Select a hairstyle", "Grooming", GLOB.hairstyles_list)
 	if(isnull(new_style))
 		return TRUE
-	if(!user.can_perform_action(src, FORBID_TELEKINESIS_REACH))
-		return TRUE //no tele-grooming
 	if(HAS_TRAIT(hairdresser, TRAIT_BALD))
-		to_chat(hairdresser, span_notice("If only growing back hair were that easy for you..."))
+		to_chat(hairdresser, span_notice("If only growing back hair were that easy for you... The reminder makes you feel terrible."))
+		hairdresser.add_mood_event("bald_hair_day", /datum/mood_event/bald)
 		return TRUE
 
 	hairdresser.set_hairstyle(new_style, update = TRUE)
 
-/obj/structure/mirror/examine_status(mob/user)
+/obj/structure/mirror/proc/change_name(mob/living/carbon/human/user)
+	var/newname = sanitize_name(tgui_input_text(user, "Who are we again?", "Name change", user.name, MAX_NAME_LEN), allow_numbers = TRUE) //It's magic so whatever.
+	if(!newname)
+		return TRUE
+	user.real_name = newname
+	user.name = newname
+	if(user.dna)
+		user.dna.real_name = newname
+	if(user.mind)
+		user.mind.name = newname
+
+// Erm ackshually the proper term is species. Get it right??
+/obj/structure/mirror/proc/change_race(mob/living/carbon/human/racist)
+	var/racechoice = tgui_input_list(racist, "What are we again?", "Race change", selectable_races)
```

Comment:
again snake case so `race_choice`. maybe the title of the tgui alert here should be "Species Change" instead since race is pretty deprecated as a term